source/common/network/utility.cc

```diff
@@ -156,6 +156,16 @@ Address::InstanceConstSharedPtr Utility::parseInternetAddress(const std::string&
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+Address::InstanceConstSharedPtr
+Utility::parseInternetAddressAndPortNoThrow(const std::string& ip_address, bool v6only) {
```

Comment:
Can you flip this so the main logic is in the no throw version and then the throw version just checks the return value and throws? I think that would be better in terms of no exceptions on the data path.